cinema/models.py

```diff
@@ -125,22 +166,94 @@ class Meta:
         ordering = ["name"]
 
 
+class CinemaPersonManager(models.Manager):
+    """
+    Custom Manager, adding extra manager methods.
+
+    Custom Manager used in 'CinemaPerson' model by extending base Manager class
+    and instantiating custom Manager in 'CinemaPerson' model.
+    """
+    def get_queryset(self):
+        """
+        Return QuerySet object with cached data of some related models.
+        """
+        return super().get_queryset().select_related("user", "country")
+
+    def get_brief_persons_data(self):
+        """
+        Return QuerySet object with data of specified fields of
+        'CinemaPerson' and 'User' models.
+        """
+        return super().get_queryset().select_related("user").values_list(
+            "pk", "user__first_name",  "user__last_name", named=True
+        )
+
+    def filter_by_search_word(self, search_word):
+        """
+        Return QuerySet object, value of selected field corresponds to
+        search word entered by visitor.
+        """
+        split_query = search_word.split()
+        q = (Q(user__first_name__istartswith=split_query[0]) &
+             Q(user__last_name__iendswith=split_query[-1])) | Q(
+            user__first_name__icontains=search_word) | Q(
+            user__last_name__icontains=search_word
+        )
+        return self.get_queryset().filter(q)
+
+    def get_cached_persons_data(self):
+        """
+        Return QuerySet object with cached data of all related models.
+        """
+        return self.get_queryset().prefetch_related(
+            "news_set", "film_set__genre",
+            Prefetch("film_set", Film.objects.select_related("imdb_rating"))
+        )
+
+    def get_specified_persons_data(self, fields_names):
+        """
+        Return QuerySet object with data of specified fields of
+        'CinemaPerson' model and some related models.
+        """
+        return self.get_cached_persons_data().values_list(
+            "pk", *fields_names, "news__title", named=True
+        ).order_by("-news__created_at")
+
+
 class CinemaPerson(ExtendedProfileMixin):
+    """
+    Save data about actors, directors, writers needed for 'Film' model.
+
+    Model related with next models:
+    - User (one-to-one relationship);
+    - Film, News (many-to-many relationship);
+    - CommentToPerson (many-to-one relationship).
+
+    Class of this model inherits from 'ExtendedProfileMixin' class.
+    """
     user = models.OneToOneField(User, on_delete=models.PROTECT)
     bio = models.TextField(unique=True)
     oscar_awards = models.PositiveSmallIntegerField(default=0)
     avatar = models.ImageField(
         upload_to=upload_to_cinema_person, blank=True, null=True
     )
+    objects = models.Manager()
```

Comment:
Do we really need it? I think it will be good without the definition.

Reply:
Without this case nothing work